File: ruby/test/test_teams.rb
# frozen_string_literal: true

require_relative './base_template'

# All tests related to Onvo's teams endpoints
class TeamsTest < BaseTemplate
  def test_team_list
    assert_silent { @onvo.teams.list }
  end

  def test_team_get
    sample_test_id = @onvo.teams.list[0]['id']
    assert_silent { @onvo.teams.get(sample_test_id) }
  end

  def test_update
    sample_test_id = @onvo.teams.list[0]['id']
    assert_silent { @onvo.teams.update(sample_test_id, { 'name': 'Onvo AI', 'phone_number': '+91 807 506 5108' }) }
  end
end

File: python/tests/test_teams.py
from .base_test import BaseTest


class TestTeams(BaseTest):

    def test_list(self):
        self.assertShouldRaise(None, self.onvoSDK.teams.list)

    def test_get(self):
        sample_id = self.onvoSDK.teams.list()[0]["id"]
        self.assertShouldRaise(None, lambda: self.onvoSDK.teams.get(sample_id))

    def test_update(self):
        sample_id = self.onvoSDK.teams.list()[0]["id"]
        self.assertShouldRaise(
            None,
            lambda: self.onvoSDK.teams.update(
                sample_id,
                {"name": "Onvo AI", "phone_number": "+91 807 506 5108"},
            ),
        )
